one-pager-maker/src/pages/SignUp.tsx

```diff
@@ -23,24 +29,50 @@ const SignUp = () => {
 
     const createUser = useCallback(() => {
 
+        // user id validation
+        try {
+            assertZodSchema(idSchema, formData.userId)
+        } catch {
+            alert(`User ID is allowed only alphanumeric characters, underscores (_), and hyphens (-).`);
+            return
+        }
+
+        if (result.data) {
+            alert(`User ID "${formData.userId}" is already registerd.`);
+            return
+        } else if (result.error) {
+            alert(`Server Internal Error. Please retry.`);
+            return
+        }
+
+        // Regist user
```

Comment:
Similar code found in SetID.tsx, so please add comment to remember refactor this.
Remember: you dont have to fix it right now.

I think that this should be done in application service layer ideally.

Reply:
Sorry. I added TODO comment.